modules/mini-games/Race.lua

```diff
@@ -252,177 +279,177 @@ local function insideBox(box, pos)
+            game.print(lap_format:format(name, lap_time, laps[name], variables["laps"]))
+            scores[name].time = game.tick
+
+            -- Add the lap to the total
+            if scores[name].total_time then
+                scores[name].total_time = math.round(scores[name].total_time + lap_time, 4)
+            else
+                scores[name].total_time = math.round(lap_time, 4)
+            end
+
+            -- Check if a player has completed all laps
+            if laps[name] >= variables["laps"] then
+                cars[name].destroy()
+                cars[name] = nil
+                if player.character then
+                    player.character.destroy()
+                end
+
+                -- Print and update finish times
+                game.print(finish_format:format(name, scores[name].total_time, Nth(variables["place"])))
+                scores["finish_times"][name] = { variables["place"], scores[name].total_time }
+                variables["place"] = variables["place"] + 1
+
+                -- If all players have finished then end the game
+                if variables["place"] > #game.connected_players - variables["new_joins"] then
+                    Mini_games.stop_game()
+                end
+            end
+        end
     end
 end
 
-local function invisabilty(car,name)
+--- Make the car indestructible and cause all biters to flee
+local function start_invincibility(car,name)
     car.destructible = false
     local biters = surface[1].find_enemy_units(variables["Dead_car"][name].position, 50, "player")
     for i, biter in ipairs(biters) do
-        biter.set_command({type = defines.command.flee, distraction  = defines.distraction.by_anything, from = car})
+        biter.set_command{ type = defines.command.flee, distraction = defines.distraction.by_anything, from = car }
     end
 end
 
-local token_for_stop_invins = Token.register(stop_invins)
-
-local token_for_kill_biters = Token.register(kill_biters)
+--- Make the car and player destructible again
+local stop_invincibility = Token.register(function(name)
+    variables["Dead_car"][name].car.destructible = true
+    local player = variables["Dead_car"][name].player
+    player.character.destructible = true
+end)
 
+--- Kill all biters in a close range to the player
+local kill_biters = Token.register(function(name)
+    local biters = surface[1].find_enemy_units(variables["Dead_car"][name].position, 3, "player")
+    for i, biter in ipairs(biters) do
+        biter.destroy()
+    end
+end)
 
-local respawn_car = function(name)
+--- Respawn the car for a player
+local respawn_car = Token.register(function(name)
     local player = variables["Dead_car"][name].player
-    local car =
-        surface[1].create_entity {
+    local position = surface[1].find_non_colliding_position('car', variables["Dead_car"][name].position, 5, 0.5)
+    position = position or variables["Dead_car"][name].position
```

Comment:
does this mean that if theirs no position it will take itself? as if this is so then then wont it spawn inside onther car

Reply:
There is a very slim chance that no position is found, but in the case non is found you would rather it spawn in another car than not spawn at all leaving the player unable to compete. An alternative could be to repeat the spawn car task until a position is found.